packages/components/src/forms/form-input-validation/index.tsx

```diff
@@ -0,0 +1,48 @@
+import classNames from 'classnames';
+import { ReactNode } from 'react';
+import { Gridicon } from '../..';
+
+import './style.scss';
+
+interface Props {
+	isError: boolean;
+	isWarning?: boolean;
+	isHidden?: boolean;
+	isValid?: boolean;
+	text: ReactNode;
+	icon?: string;
+	id?: string;
+	className?: string;
+	children?: ReactNode;
+}
+
+const FormInputValidation: React.FC< Props > = ( {
+	isError = false,
+	isWarning,
+	isHidden,
+	className,
+	text,
+	icon,
+	id,
+	children,
+} ) => {
+	const classes = classNames( className, {
+		'form-input-validation': true,
+		'is-warning': isWarning,
+		'is-error': isError,
+		'is-hidden': isHidden,
+	} );
+
+	const defaultIcon = isError || isWarning ? 'notice-outline' : 'checkmark';
+
+	return (
+		<div className={ classes } role="alert">
+			<span id={ id }>
+				<Gridicon size={ 24 } icon={ icon ? icon : defaultIcon } /> { text }
+				{ children }
+			</span>
```

Comment:
I think these two (icon and childen) need to be in their own elements to fix this vertical alignment issue:

<img width="407" alt="image" src="https://user-images.githubusercontent.com/17054134/169083287-ec0f8ef4-4071-42ec-a29a-52600672f31e.png">